lib/utils/findMediaFeatureNamesSearch.js

```diff
@@ -0,0 +1,147 @@
+	isMediaQueryInvalid,
+	parseFromTokens,
+} = require('@csstools/media-query-list-parser');
+
+/** @typedef {Array<import('@csstools/media-query-list-parser').MediaQuery>} MediaQueryList */
+/** @typedef {import('@csstools/css-tokenizer').TokenIdent} TokenIdent */
+/** @typedef {{ stringify: () => string }} MediaQuerySerializer */
+
+const rangeFeatureOperator = /[<>=]/;
+
+/**
+ * Search a CSS string for Media Feature names.
+ * For every found name, invoke the callback, passing the token
+ * as an argument.
+ *
+ * Found tokens are mutable and modifications made to them will be reflected in the output.
+ *
+ * This function supports some non-standard syntaxes like SCSS variables and interpolation.
+ *
+ * @param {string} mediaQueryParams
+ * @param {(mediaFeatureName: TokenIdent) => void} callback
+ *
+ * @returns {MediaQuerySerializer}
+ */
+module.exports = function findMediaFeatureNamesSearch(mediaQueryParams, callback) {
+	const tokens = tokenize({ css: mediaQueryParams });
+	const list = parseCommaSeparatedListOfComponentValues(tokens);
+
+	const mediaQueryConditions = list.flatMap((listItem) => {
+		return listItem.flatMap((componentValue) => {
+			if (
+				!isSimpleBlockNode(componentValue) ||
+				componentValue.startToken[0] !== TokenType.OpenParen
+			) {
+				return [];
+			}
+
+			const blockTokens = componentValue.tokens();
+
+			const mediaQueryList = parseFromTokens(blockTokens, {
+				preserveInvalidMediaQueries: true,
+			});
+
+			return mediaQueryList.filter((mediaQuery) => {
+				return !isMediaQueryInvalid(mediaQuery);
+			});
+		});
+	});
+
+	mediaQueryConditions.forEach((mediaQuery) => {
+		mediaQuery.walk(({ node }) => {
+			if (isMediaFeature(node)) {
+				const token = node.getNameToken();
+
+				if (token[0] !== TokenType.Ident) return;
+
+				callback(token);
+			}
+
+			if (isGeneralEnclosed(node)) {
+				const relevantTokens = topLevelTokenNodes(node);
+
+				if (!relevantTokens) {
+					return;
+				}
+
+				relevantTokens.forEach((token, i) => {
+					if (token[0] !== TokenType.Ident) {
+						return;
+					}
+
+					const nextToken = relevantTokens[i + 1];
+					const prevToken = relevantTokens[i - 1];
+
+					if (
+						// Media Feature
+						(!prevToken && nextToken && nextToken[0] === TokenType.Colon) ||
+						// Range Feature
+						(nextToken &&
+							nextToken[0] === TokenType.Delim &&
+							rangeFeatureOperator.test(nextToken[4].value)) ||
+						// Range Feature
+						(prevToken &&
+							prevToken[0] === TokenType.Delim &&
+							rangeFeatureOperator.test(prevToken[4].value))
+					) {
+						callback(token);
+					}
+				});
+			}
+		});
+	});
+
+	return {
+		stringify() {
+			return stringify(...tokens);
+		},
+	};
+};
+
+/** @param {import('@csstools/media-query-list-parser').GeneralEnclosed} node */
+function topLevelTokenNodes(node) {
+	const components = node.value.value;
+
+	if (isToken(components) || components.length === 0 || isToken(components[0])) {
+		return false;
```

Comment:
[suggest] How about simplifying the return type of this function?

```suggestion
		return [];
```

Reply:
Yes, this better, thanks!